src/components/TodoItem.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <li class="list-group-item d-flex justify-content-between">
+    <div>
+      <input
+        type="checkbox"
+        class="mr-3"
+        required
+        :checked="todo.completed"
+        @change="handleToggleCompleted(todo)"
+      />
+      <span>{{ todo.title }}</span>
+    </div>
+    <div class="my-n2">
+      <button class="btn px-1 mr-1" @click="handleClickEdit(todo)">
+        <fa icon="edit" class="edit-icon"></fa>
+      </button>
+      <button class="btn px-1" @click="handleClickRemove(todo)">
+        <fa icon="trash" class="trash-icon"></fa>
+      </button>
+    </div>
+  </li>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+import { FontAwesomeIcon } from "@/plugins/font-awesome";
+import Todo from "@/models/todo";
+
+export type HandleToggleCompleted = (todo: Todo) => void;
+export type HandleClickEdit = (todo: Todo) => void;
+export type HandleClickRemove = (todo: Todo) => void;
+
+export default defineComponent({
+  components: { fa: FontAwesomeIcon },
```

Comment:
好み

自分なら↓みたいに書いて、コンポーネントと通常のhtmlタグを視覚的にも区別しやすくします。
あとそっちのほうがgrepしやすい気もするけど、実際はそんなに大差ないかも。

```suggestion
  components: { FontAwesomeIcon },
```

```
<FontAwesomeIcon icon="trash" />
```

Reply:
kebabケースで書く場合は、ハイフン入りの名前が推奨と公式ガイドにあったはずです。

https://jp.vuejs.org/v2/style-guide/index.html#%E3%83%86%E3%83%B3%E3%83%97%E3%83%AC%E3%83%BC%E3%83%88%E5%86%85%E3%81%A7%E3%81%AE%E3%82%B3%E3%83%B3%E3%83%9D%E3%83%BC%E3%83%8D%E3%83%B3%E3%83%88%E5%90%8D%E3%81%AE%E5%BD%A2%E5%BC%8F-%E5%BC%B7%E3%81%8F%E6%8E%A8%E5%A5%A8

Reply:
修正しました。